chromatic_fitting.py

```diff
@@ -381,76 +542,112 @@ def __init__(self, **kw):
+            models = self.pymc3_model
+            datas = [self.get_data(i) for i in range(self.data.nwave)]
+            orbits = self.orbit
+        else:
+            models = [self.pymc3_model]
+            datas = [self.get_data()]
+            orbits = [self.orbit]
 
-                # self.every_light_curve = dict(Counter(self.every_light_curve)+Counter({f"wavelength_{i}":mu}))
-                if f"wavelength_{i}" not in self.every_light_curve.keys():
-                    self.every_light_curve[f"wavelength_{i}"] = pm.math.sum(
-                        light_curves, axis=-1
-                    ) + (self.parameters["baseline"].get_prior(i))
-                else:
-                    self.every_light_curve[f"wavelength_{i}"] += pm.math.sum(
+        if not hasattr(self, 'every_lightcurve'):
+            self.every_light_curve = {}
+
+        for j, (mod, data, orbit) in enumerate(zip(models, datas, orbits)):
+            with mod:
+                for i, w in enumerate(data.wavelength):
+                    # create quadratic limb-darkening lightcurves from Exoplanet
+                    limb_darkening = self.parameters["limb_darkening"].get_prior(j+i)
+                    planet_radius = self.parameters["radius_ratio"].get_prior(j+i) *\
+                                    self.parameters["stellar_radius"].get_prior(j+i)
+                    light_curves = xo.LimbDarkLightCurve(limb_darkening).get_light_curve(
+                        orbit=orbit,
+                        r=planet_radius,
+                        t=list(data.time.to_value("day")),
+                    )
+
+                    # calculate the transit + baseline model
+                    mu = pm.math.sum(
                         light_curves, axis=-1
-                    ) + (self.parameters["baseline"].get_prior(i))
+                    ) + (self.parameters["baseline"].get_prior(j+i))
+
+                    # self.every_light_curve = dict(Counter(self.every_light_curve)+Counter({f"wavelength_{i}":mu}))
+                    if f"wavelength_{j+i}" not in self.every_light_curve.keys():
+                        self.every_light_curve[f"wavelength_{j+i}"] = pm.math.sum(
+                            light_curves, axis=-1
+                        ) + (self.parameters["baseline"].get_prior(j+i))
+                    else:
+                        self.every_light_curve[f"wavelength_{j+i}"] += pm.math.sum(
+                            light_curves, axis=-1
+                        ) + (self.parameters["baseline"].get_prior(j+i))
 
-            self.model_chromatic_transit_flux = [
-                self.every_light_curve[k] for k in tqdm(self.every_light_curve)
-            ]
-            # pm.Deterministic("model_chromatic_flux", self.model_chromatic_flux)
+                self.model_chromatic_transit_flux = [
+                    self.every_light_curve[k] for k in tqdm(self.every_light_curve)
+                ]
 
     def run_simultaneous_fit(self, r):
```

Comment:
Could/should we remove `setup_orbit` from this wrapper, and then move it up to be shared by all models?

Reply:
Yep!